lib/fog/openstack/core.rb

```diff
@@ -169,13 +173,113 @@ def self.authenticate_v2(options, connection_options = {})
       }
     end
 
+    # Keystone Style Auth
+    def self.authenticate_v3(options, connection_options = {})
+      uri                   = options[:openstack_auth_uri]
+      tenant_name           = options[:openstack_tenant]
+      service_type          = options[:openstack_service_type]
+      service_name          = options[:openstack_service_name]
+      identity_service_type = options[:openstack_identity_service_type]
+      endpoint_type         = (options[:openstack_endpoint_type] || 'public').to_s
+      openstack_region      = options[:openstack_region]
+      domain_name           = options[:openstack_domain]
```

Comment:
I have tested this code, but it is failing in core.rb, as user domain and project domain should not be necessarily the same. (line 340, 350)

This certainly means that I need to split project and user domain. 

I can do the update and propose a merge request if you want.